libs/src/LSPlugin.ts

```diff
@@ -75,20 +73,34 @@ export interface LSPluginPkgConfig {
   mode: 'shadow' | 'iframe'
   themes: Theme[]
   icon: string
-
-  [key: string]: any
+  /**
+   * Alternative entrypoint for development.
+   */
+  devEntry: unknown
+  /**
+   * For legacy themes, do not use.
+   */
+  theme: unknown
 }
 
 export interface LSPluginBaseInfo {
-  id: string // should be unique
+  /**
+   * Must be unique.
+   */
+  id: string
   mode: 'shadow' | 'iframe'
-
   settings: {
     disabled: boolean
-    [key: string]: any
-  }
-
-  [key: string]: any
+  } & Record<string, unknown>
```

Comment:
I realised after putting up this PR that Plugins can define their own settings as they go. Making this Generic is probably a better idea, but we can make iterative improvements: for now, indicating that `settings` definitely contains `disabled` and may contain any other string as an unknown is a safer way of typing it.

Why is `unknown` better than `any`? `any` can be _any type_, while `unknown` is not any specific type and _must_ be narrowed down to be used safely (or overwritten). [Here's another demo](https://tsplay.dev/wg8k6N)